File: front/admin_web/src/api/goods.ts
import { httpHeader, AxiosPromise } from "./api"

export let getGoodslist = (page: string, page_size: string, author_id: number, warehouse_id: number, start_date: string, end_date: string, name: string): AxiosPromise => {
    let url = `/goods/get-list?page=${page}&page_size=${page_size}`
    if (author_id != 0) {
        url = url + `&author_id=${author_id}`
    }
    if (warehouse_id != 0) {
        url = url + `&warehouse_id=${warehouse_id}`
    }
    if (start_date != "") {
        url = url + `&start_date=${start_date}&end_date=${end_date}`
    }
    if (name != "") {
        url = url + `&name=${name}`
    }
    return httpHeader("GET", url, {}, {})
}

// 定义保存商品参数类型
export interface SaveGoodsParams {
    warehouse_id: number,
    cost_price: number,
    supply_price: number,
    sale_number: number,
    return_number: number,
    link_return_number: number,
    name: string,
    date: string,
    unit: string
    id: number,
}

// 保存
export let saveGoods = (params: SaveGoodsParams): AxiosPromise => {
    return httpHeader("POST", "/goods/save", params, {})
}

// 定义修改商品参数类型
export interface ModifyGoodsParams {
    id: number,
    warehouse_id: number,
    cost_price: number,
    supply_price: number,
    sale_number: number,
    return_number: number,
    link_return_number: number,
    name: string,
    date: string,
    unit: string
}

// 修改
export let modifyGoods = (params: SaveGoodsParams): AxiosPromise => {
    return httpHeader("POST", "/goods/modify", params, {})
}

// 删除
export let deleteGoods = (id: number): AxiosPromise => {
    let params = {
        "id": id
    }
    return httpHeader("POST", "/goods/delete", params, {})
}

File: back/admin_server/web/handler/goods.go
package handler

import (
	"strconv"

	"goods/common/constant"
	"goods/common/util"
	"goods/service"
	"goods/web/form"
	"goods/web/middleware"

	"github.com/gin-gonic/gin"
)

type Goods struct{}

func (g *Goods) SaveOne(c *gin.Context) {
	user := middleware.GetUser(c)
	ctx := util.ContextTracer(c)
	params := c.MustGet(constant.GinBindFormKey).(*form.GoodsSaveOneForm)
	if err := new(service.Goods).SaveOne(ctx, params.WarehouseId, user.Id, params.Name, params.Date, params.Unit, params.CostPrice, params.SupplyPrice, params.SaleNumber, params.ReturnNumber, params.LinkReturnNumber); err != nil {
		util.Fail(c, err)
		return
	}
	util.Success(c, gin.H{})
}

func (g *Goods) ModifyOne(c *gin.Context) {
	user := middleware.GetUser(c)
	ctx := util.ContextTracer(c)
	params := c.MustGet(constant.GinBindFormKey).(*form.GoodsModifyOneForm)
	if err := new(service.Goods).ModifyOne(ctx, params.Id, params.WarehouseId, user.Id, params.Name, params.Date, params.Unit, params.CostPrice, params.SupplyPrice, params.SaleNumber, params.ReturnNumber, params.LinkReturnNumber); err != nil {
		util.Fail(c, err)
		return
	}
	util.Success(c, gin.H{})
}

func (g *Goods) DeleteOne(c *gin.Context) {
	ctx := util.ContextTracer(c)
	params := c.MustGet(constant.GinBindFormKey).(*form.GoodsDeleteOneForm)
	if err := new(service.Goods).DeleteOne(ctx, params.Id); err != nil {
		util.Fail(c, err)
		return
	}
	util.Success(c, gin.H{})
}

func (g *Goods) GetList(c *gin.Context) {
	ctx := util.ContextTracer(c)
	// 获取参数
	page, _ := strconv.Atoi(c.DefaultQuery("page", "1"))
	pageSize, _ := strconv.Atoi(c.DefaultQuery("page_size", "10"))
	authorId, _ := strconv.Atoi(c.DefaultQuery("author_id", "0"))
	warehouseId, _ := strconv.Atoi(c.DefaultQuery("warehouse_id", "0"))
	startDate := c.DefaultQuery("start_date", "")
	endDate := c.DefaultQuery("end_date", "")
	name := c.DefaultQuery("name", "")
	data, err := new(service.Goods).GetList(ctx, int32(page), int32(pageSize), int32(authorId), int32(warehouseId), startDate, endDate, name)
	if err != nil {
		util.Fail(c, err)
		return
	}
	util.Success(c, data)
}
